pkg/roachprod/cloud/cluster_cloud.go

```diff
@@ -155,9 +179,43 @@ func (c *Cluster) PrintDetails(logger *logger.Logger) {
 	} else {
 		logger.Printf("(no expiration)")
 	}
+	logPrettifiedHeader(logger, printDetailsColumnHeaders)
 	for _, vm := range c.VMs {
-		logger.Printf("  %s\t%s\t%s\t%s\t%s\t%s\t%s", vm.Name, vm.DNS, vm.PrivateIP, vm.PublicIP, vm.MachineType, vm.CPUArch, vm.CPUFamily)
+		provisionModel := onDemandProvisionModel
+		if vm.Preemptible {
+			provisionModel = spotProvisionModel
+		}
+		logger.Printf(prettifyRow(printDetailsColumnHeaders, map[string]string{
+			headerName: vm.Name, headerDNS: vm.DNS, headerPrivateIP: vm.PrivateIP, headerPublicIP: vm.PublicIP,
+			headerMachineType: vm.MachineType, headerCPUArch: string(vm.CPUArch), headerCPUFamily: vm.CPUFamily,
+			headerProvisionModel: provisionModel,
+		}))
+	}
+}
+
+// logPrettifiedHeader logs a prettified row of headers.
+func logPrettifiedHeader(logger *logger.Logger, headers []string) {
+	row := ""
+	for _, header := range headers {
+		row = fmt.Sprintf("%s%s\t", row, header)
```

Comment:
Nit: I've run `roachprod list` by itself (without details) and `tabwriter.NewWriter` seems to do a better job of aligning the columns [1]. On my system the current print out is misaligned.
[1] https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachprod/main.go#L322